src/script/interpreter.cpp

```diff
@@ -2535,6 +2661,12 @@ unsigned int GenericTransactionSignatureChecker<T>::GetTxVersion() const
     return txTo->nVersion;
 }
 
+template <class T>
+unsigned int GenericTransactionSignatureChecker<T>::GetTxWeight() const
+{
+    return ::GetSerializeSize(*txTo, PROTOCOL_VERSION | SERIALIZE_TRANSACTION_NO_WITNESS) * (WITNESS_SCALE_FACTOR - 1) + ::GetSerializeSize(*txTo, PROTOCOL_VERSION);
+}
+
```

Comment:
looks like this is just copied from `GetTranasactionWeight` in src/validation.h .... can you directly call that method rather than reimplementing it?

Reply:
I could not do it because the function is defined on a concrete type `CTransaction` and we are using a generic type name here.

Reply:
Can you try casting by passing `CTransaction(*txTo)`?

Reply:
Hmm, the compiler should let you do this

Reply:
Yep, it does.

Reply:
The more I think the *txTo might be creating a copy.

Reply:
No, `GetSerializeSize` takes its input by reference. There is no copy.